src/twisted/internet/test/test_process.py

```diff
@@ -844,6 +845,59 @@ def spawnChild():
 
         reactor.callWhenRunning(spawnChild)
         self.runReactor(reactor)
+
+
+    @onlyOnPOSIX
+    def test_process_unregistered_before_protocol_ended_callback(self):
+        """
+        Process is removed from reapProcessHandler dict before running
+        ProcessProtocol.processEnded() callback.
+        """
+        results = []
+
+        class TestProcessProtocol(ProcessProtocol):
```

Comment:
All of these inner classes and functions [need docstrings](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html#docstrings).